pre-registration/pre-registration-application-service/src/main/java/io/mosip/preregistration/application/exception/util/PreRegistrationExceptionHandler.java

```diff
@@ -502,7 +503,8 @@ public ResponseEntity<ResponseWrapper<ServiceError>> methodArgumentNotValidExcep
 	public ResponseEntity<ResponseWrapper<ServiceError>> onHttpMessageNotReadable(
 			final HttpServletRequest httpServletRequest, final HttpMessageNotReadableException e) throws IOException {
 		ResponseWrapper<ServiceError> errorResponse = setErrors(httpServletRequest);
-		ServiceError error = new ServiceError(ErrorCodes.PRG_CORE_REQ_015.getCode(), e.getMessage());
+		String errorMessage = StringUtils.substringBefore(e.getMessage(), ":");
```

Comment:
I think this is not the correct way to solve the MOSIP-23430 issue, if the Message has some : in the message itself, then we may be given half message using this approach.